aggregator/src/main/java/com/redhat/cloud/notifications/db/EndpointEmailSubscriptionResources.java

```diff
@@ -14,13 +15,14 @@
 
     @Transactional
     public Long getEmailSubscribersCount(String accountNumber, String bundleName, String applicationName, EmailSubscriptionType subscriptionType) {
-        String query = "SELECT COUNT(id.userId) FROM EmailSubscription WHERE id.accountId = :accountId " +
-                "AND application.bundle.name = :bundleName AND application.name = :applicationName AND id.subscriptionType = :subscriptionType";
-        return session.createQuery(query, Long.class)
+        String query = "SELECT COUNT(user_id) FROM endpoint_email_subscriptions es, applications a, bundles b WHERE es.account_id = :accountId " +
+                "AND es.application_id = a.id AND a.bundle_id = b.id AND b.name = :bundleName AND a.name = :applicationName AND es.subscription_type = :subscriptionType";
+        BigInteger result = (BigInteger) session.createNativeQuery(query)
                 .setParameter("accountId", accountNumber)
                 .setParameter("bundleName", bundleName)
                 .setParameter("applicationName", applicationName)
-                .setParameter("subscriptionType", subscriptionType)
+                .setParameter("subscriptionType", subscriptionType.name())
                 .getSingleResult();
+        return result.longValue();
```

Comment:
Normally, converting a `BigInteger` to a long value would be an issue because of the potential data loss, but we're not actually using that value in the calling method. Sonar might be unhappy because of that.

Reply:
We have a Sonar running?! I didn't know that! Please send me a link :)

Reply:
It's well hidden behind the VPN. I didn't bookmark the link but I think you can find it somewhere in the Jenkins build log.